### ISSUE TITLE
error and coredump on fork

### ISSUE THREAD

**Issue by cc32d9** [EOSChronicleProject/eos-chronicle#6]
```
Aug 31 05:03:51 chronicle chronicle-receiver[73]: info  2019-08-31T05:03:51.163 chronicle receiver_plugin.cpp:643       receive_result       ] fork detected at block 76859237; head=76859246
Aug 31 05:03:51 chronicle chronicle-receiver[73]: error 2019-08-31T05:03:51.213 chronicle receiver_plugin.cpp:1337      ack_block            ] Exporter acked block=76859236, but block=76859237 was already acknowledged
Aug 31 05:03:51 chronicle chronicle-receiver[73]: Exporter acked block below prevuously acked one
Aug 31 05:03:51 chronicle systemd[1]: chronicle_receiver@eos.service: Main process exited, code=dumped, status=11/SEGV
Aug 31 05:03:51 chronicle systemd[1]: chronicle_receiver@eos.service: Failed with result 'core-dump'.
```

**Comment by cc32d9** [comment]
on another server, same error, but process exited cleanly and database didn't get dirty

**Comment by cc32d9** [comment]
fixed the error message in 7cacd5d5

**Comment by cc32d9** [comment]
coredump was likely because of socket handling in Boost. There were few issues that should be fixed in the next release.
